flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -112,8 +120,11 @@
 
     // ------------------------------------------------------------------------
 
-    PipelinedSubpartition(int index, ResultPartition parent) {
+    PipelinedSubpartition(int index, int buffersPerChannel, ResultPartition parent) {
         super(index, parent);
+
+        checkArgument(buffersPerChannel >= 0, "Buffers per channel must be non-negative.");
+        this.buffersPerChannel = buffersPerChannel;
```

Comment:
Can we rename this property to something like `receiverExclusiveBuffersPerChannel`?  Because actually this is not the number of `buffersPerChannel` for the sender.